packages/dev/core/src/Materials/Node/Blocks/PBR/clearCoatBlock.ts

```diff
@@ -255,6 +264,11 @@ export class ClearCoatBlock extends NodeMaterialBlock {
         if (ccBlock) {
             state._emitUniformFromString("vClearCoatRefractionParams", "vec4");
             state._emitUniformFromString("vClearCoatTangentSpaceParams", "vec2");
+            
+            const normalShading = ccBlock.worldNormal;
+            code += `vec3 vGeometricNormaClearCoatW = ${normalShading.isConnected ? "normalize(" + normalShading.associatedVariableName + ".xyz)" : "geometricNormalW"};\r\n`;

```

Comment:
You should ask the state to give you a name to avoid collisions

Reply:
Discussed offline, this prevents it https://github.com/BabylonJS/Babylon.js/pull/12388/files#diff-60b6c02985dd42f81be35afcb88e7045f907548a1796e4b1b129d53776424abeR72